uniffi/src/interface/mod.rs

```diff
@@ -466,6 +487,17 @@ impl APIConverter<Argument> for weedle::argument::SingleArgument<'_> {
     }
 }
 
+impl APIConverter<Error> for weedle::common::Identifier<'_> {
+    fn convert(&self, ci: &ComponentInterface) -> Result<Error> {
+        Ok(Error {
+            name: self.0.to_string(),
+            type_: ci
+                .get_type_definition(self.0)
```

Comment:
Careful, just because the error type isn't known at the time you encounter it in the parse tree, doesn't mean it's invalid. It's OK for the consumer to have done something like:

```
interface Example {
  [Throws=Error]
  void example();
};

enum Error {
  "EXAMPLE"
};
```

There probably needs to be an entirely separate "consistency checking" phase where we walk the parsed `ComponentInterface` and check that all return types and defined, there isn't an enum and a record with the same name, etc. Right now we only catch these errors by virtual of *generating invalid rust scaffolding*, which is not a great developer experience.

But I don't think you should have to tackle that inline in this PR.

Reply:
> just because the error type isn't known at the time you encounter it in the parse tree, doesn't mean it's invalid. 

Hrm, sorry, I got a bit confused here - we already have a separate "type finding" step that will handle the use-before-definition problem I mention above, although there are plenty of other consistency checks that it *doesn't* perform. If you decide to keep the current approach then I think you want `resolve_type_definition` rather than `get_type_definition` here. (Which I realize there's not really any way you could know that because this is not well documented so, that's something for me to take a look at adding).

I still feel like you want a `TypeReference` of some sort here rather than a separate struct for errors. But looking that `TypeReference` up by name using `resolve_type_definition` would be the right way to obtain one; see e.g. how we resolve the return type when construction a `Function` instance.

Reply:
Hmmm ended up storing the full list of attributes in the method here, then later would check if an error exists in there. Seems like the compiler errors out semi-nicely with that if something is missing